hs_core/models.py

```diff
@@ -2036,18 +2036,22 @@ async def update_record(self, record):
                         record.agency_name = agency_name_new
                         record.agency_url = item['id']
                         await sync_to_async(record.save)()
-                        await update_crossref_meta_deposit.apply_async((self.short_id,))
+                        metadata = record.metadata
```

Comment:
you're headed in the right direction here!
But `record.metadata` is going to give you an error
Here your "record" is a FundingAgency object.
[FundingAgency](https://github.com/hydroshare/hydroshare/blob/2.15.9/hs_core/models.py#L1979) class does not have any "metadata" attribute.

```
Running migrations:
  Applying hs_core.0086_auto_20241022_1602_crossref_to_ror_migration.../usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1063: InsecureRequestWarning: Unverified HTTPS request is being made to host 'api.ror.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1063: InsecureRequestWarning: Unverified HTTPS request is being made to host 'api.ror.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py:1063: InsecureRequestWarning: Unverified HTTPS request is being made to host 'api.ror.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
Traceback (most recent call last):
  File "/hydroshare/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 106, in wrapper
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/migrate.py", line 356, in handle
    post_migrate_state = executor.migrate(
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 135, in migrate
    state = self._migrate_all_forwards(
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
    state = self.apply_migration(
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 252, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/migration.py", line 132, in apply
    operation.database_forwards(
  File "/usr/local/lib/python3.9/site-packages/django/db/migrations/operations/special.py", line 193, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/hydroshare/hs_core/models.py", line 2054, in migrate_from_crossref_to_ror
    loop.run_until_complete(cls.get_funder_records(apps))
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/hydroshare/hs_core/models.py", line 2048, in get_funder_records
    await asyncio.gather(*(cls.update_record(record) for record in funding_records))
  File "/hydroshare/hs_core/models.py", line 2039, in update_record
    metadata = record.metadata
AttributeError: 'FundingAgency' object has no attribute 'metadata'
```

Reply:
Please test your migration against real data before requesting review again.
You can do that by creating a resource on `develop` branch and then checking out this PR branch and running `./hsctl managepy migrate`

Reply:
You can get the resource short_id for a given funding agency by using the [AbstractMetadataElement.content_object](https://github.com/hydroshare/hydroshare/blob/2.15.9/hs_core/models.py#L450)

```py
from hs_core.models import FundingAgency
fa = FundingAgency.objects.first()
resource_id = fa.content_object.resource.short_id
```